x-pack/plugins/enterprise_search/server/lib/indices/create_api_key.ts

```diff
@@ -17,14 +17,22 @@ export const createApiKey = async (
   indexName: string,
   keyName: string
 ) => {
+  let aclIndexName: string;
+  if (indexName.startsWith('search-')) {
+    const suffix = indexName.replace('search-', '');
```

Comment:
why do we have to replace the prefix?

Reply:
I'm wondering the same thing. Seems to me that this could create conflicting index names.

Reply:
Right, my question is really: why *don't* we want it to be `.search-acl-filter-search-test`?

Reply:
because then you have `search` in the prefix twice and it's redundant

Reply:
Not really? The second search isn't part of the prefix, it's the actual index name that just happens to start with search. What happens when people want to use their own indices that don't start with `search` (which we don't really support, but people can easily do)? We may accidentally create conflicting ACL index names (eg: `search-test` and `test` would have the same ACL name).

Reply:
++ @sphilipse - I am not too worried about the redundancy, more about consistency in the future where we don't require the `search-` prefix.

Reply:
🤦 totally missed that I had replies on here when I asked again for an approve and then merged. Sorry both, did not mean to ignore this concern.

I do think that we're in an OK place right now. If we get rid of using `search-*`, we could switch to using `.acl-filter` as our prefix anyway.